syntax = "proto3";

package meshtrade.compliance.client.v1;

import "google/type/date.proto";
import "meshtrade/compliance/client/v1/company_representative_role.proto";
import "meshtrade/compliance/client/v1/natural_person.proto";
import "meshtrade/type/v1/contact_details.proto";

option go_package = "github.com/meshtrade/api/go/compliance/client/v1;clientv1";

/*
   CompanyRepresentative models an individual acting in an official capacity for a legal entity.
   This person is typically subject to KYC verification as part of the overall KYB process
   for the legal entity they represent.
*/
message CompanyRepresentative {
  /*
     Details of the natural person that is the company representative.
     This contains the core personal identity information (name, residential address,
     ID document, personal contact details, etc.) required for their individual KYC check.

     Required
  */
  meshtrade.compliance.client.v1.NaturalPerson natural_person = 1;

  /*
     The official role this person holds in relation to the company.

     Required
  */
  meshtrade.compliance.client.v1.CompanyRepresentativeRole role = 2;

  /*
     The person's job title or position within the company (e.g., "CEO", "Managing Partner").

     Optional
  */
  string position = 3;

  /*
     For UBOs and Shareholders, this specifies the percentage of ownership or voting rights.
     Should be a value between 0.0 and 100.0.
     Required by business logic if the role is ULTIMATE_BENEFICIAL_OWNER or SHAREHOLDER.

     Optional
  */
  float ownership_percentage = 4;

  /*
     The professional contact details for the representative in their capacity at the company
     (e.g., work email, work phone).

     Optional
  */
  meshtrade.type.v1.ContactDetails professional_contact_details = 5;

  /*
     The date when the person was appointed to this role.

     Optional
  */
  google.type.Date date_of_appointment = 6;
}
